src/test/clojure/xtdb/trie_test.clj

```diff
@@ -29,39 +29,78 @@
+       (sort-by :path)))
+
 (deftest test-merge-plan-with-nil-nodes-2700
-  (letfn [(->arrow-hash-trie [^VectorSchemaRoot meta-root]
-            (ArrowHashTrie. (.getVector meta-root "nodes")))]
-    (with-open [al (RootAllocator.)
-                t1-root (tu/open-arrow-hash-trie-root al [{Long/MAX_VALUE [nil 0 nil 1]} 2 nil
-                                                          {Long/MAX_VALUE 3, (time/instant->micros (time/->instant #inst "2023-01-01")) 4}])
-                log-root (tu/open-arrow-hash-trie-root al 0)
-                log2-root (tu/open-arrow-hash-trie-root al [nil nil 0 1])]
-
-      (t/is (= [{:path [1], :pages [{:seg :t1, :page-idx 2} {:seg :log, :page-idx 0}]}
-                {:path [2], :pages [{:seg :log, :page-idx 0} {:seg :log2, :page-idx 0}]}
-                {:path [3], :pages [{:seg :t1, :page-idx 4} {:seg :t1, :page-idx 3} {:seg :log, :page-idx 0} {:seg :log2, :page-idx 1}]}
-                {:path [0 0], :pages [{:seg :log, :page-idx 0}]}
-                {:path [0 1], :pages [{:seg :t1, :page-idx 0} {:seg :log, :page-idx 0}]}
-                {:path [0 2], :pages [{:seg :log, :page-idx 0}]}
-                {:path [0 3], :pages [{:seg :t1, :page-idx 1} {:seg :log, :page-idx 0}]}]
+  (with-open [al (RootAllocator.)
+              t1-root (tu/open-arrow-hash-trie-root al [{Long/MAX_VALUE [nil 0 nil 1]} 2 nil
+                                                        {Long/MAX_VALUE 3, (time/instant->micros (time/->instant #inst "2023-01-01")) 4}])
+              log-root (tu/open-arrow-hash-trie-root al 0)
+              log2-root (tu/open-arrow-hash-trie-root al [nil nil 0 1])]
+
+    (t/is (= [{:path [1], :pages [{:seg :t1, :page-idx 2} {:seg :log, :page-idx 0}]}
+              {:path [2], :pages [{:seg :log, :page-idx 0} {:seg :log2, :page-idx 0}]}
+              {:path [3], :pages [{:seg :t1, :page-idx 4} {:seg :t1, :page-idx 3} {:seg :log, :page-idx 0} {:seg :log2, :page-idx 1}]}
+              {:path [0 0], :pages [{:seg :log, :page-idx 0}]}
+              {:path [0 1], :pages [{:seg :t1, :page-idx 0} {:seg :log, :page-idx 0}]}
+              {:path [0 2], :pages [{:seg :log, :page-idx 0}]}
+              {:path [0 3], :pages [{:seg :t1, :page-idx 1} {:seg :log, :page-idx 0}]}]
+             (->> (HashTrieKt/toMergePlan [(-> (trie/->Segment (->arrow-hash-trie t1-root))
+                                               (assoc :seg :t1))
+                                           (-> (trie/->Segment (->arrow-hash-trie log-root))
+                                               (assoc :seg :log))
+                                           (-> (trie/->Segment (->arrow-hash-trie log2-root))
+                                               (assoc :seg :log2))]
+                                          nil
+                                          (TemporalBounds.))
+                  (merge-plan-nodes->path+pages))))))
+
+(deftest test-merge-plan-recency-filtering
+  (with-open [al (RootAllocator.)
```

Comment:
todos:

- [ ] comment up exactly what the test case is stressing - what's the case, what are you hoping the algo will do?
- [ ] are there other cases to consider here which might catch the algo out?